### PR TITLE
SONARJAVA-4260 add RestAssuredMockMvc assertions to list of test assertions

### PR DESCRIPTION
**Pull request by erwan-serandour**
[SONARJAVA-4260](https://sonarsource.atlassian.net/browse/SONARJAVA-4260)

Part of 

[SONARJAVA-4260]: https://sonarsource.atlassian.net/browse/SONARJAVA-4260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by erwan-serandour** [comment]
sonarqube is raising 1 issue, but for me, it is not relevant for this case and to maintain the consistency with the rest of the code it is better to keep the comment at the end of the line

File: its/autoscan/src/test/resources/autoscan/diffs/diff_S1130.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "ruleKey": "S1130",
   "hasTruePositives": true,
-  "falseNegatives": 25,
+  "falseNegatives": 26,
```
Comment (tomasz-tylenda-sonarsource) [inline its/autoscan/src/test/resources/autoscan/diffs/diff_S1130.json:4]: Why did this change? S1130 looks unrelated to tests.

File: java-checks-test-sources/default/pom.xml [hunk 2]
```diff
@@ -921,25 +927,25 @@
...
       <version>1.9.9.1</version>
       <scope>provided</scope>
-  </dependency>
-  <dependency>
-    <groupId>io.micronaut</groupId>
-    <artifactId>micronaut-http</artifactId>
-    <version>3.7.2</version>
-    <scope>provided</scope>
-	</dependency>
+    </dependency>
```
Comment (tomasz-tylenda-sonarsource) [inline java-checks-test-sources/default/pom.xml:930]: Let's avoid unrelated reformatting of the file. This also applies to line 612 and the 944-948. The reason is the it changes git blame layer and makes it more difficult to understand what was done in a PR.

File: java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java [hunk 3]
```diff
@@ -47,9 +47,10 @@ public final class UnitTestUtils {
     MethodMatchers.create().ofSubTypes("org.fest.assertions.GenericAssert", "org.fest.assertions.api.AbstractAssert").anyName().withAnyParameters().build(),
     // rest assured 2.x, 3.x, 4.x
     MethodMatchers.create().ofTypes(
-      "com.jayway.restassured.response.ValidatableResponseOptions", // restassured 2.x
-      "io.restassured.response.ValidatableResponseOptions" // restassured 3.x and 4.x
-    )
+        "com.jayway.restassured.response.ValidatableResponseOptions", // restassured 2.x
+        "io.restassured.response.ValidatableResponseOptions", // restassured 3.x and 4.x
+        "io.restassured.module.mockmvc.response.ValidatableMockMvcResponse" // spring mock mvc
```
Comment (tomasz-tylenda-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java:52]: Is it 'spring mock mvc' or 'restassured mock mvc'?

**Review by tomasz-tylenda-sonarsource — COMMENTED** [review]
There's some SonarQube issue as well. 

Feel free to just reply if everything is fine and I misunderstood something.
